src/app/private/[uid]/admin/edit-profile/EditProfileForm.tsx

```diff
@@ -0,0 +1,100 @@
+"use client";
+
+import React from "react";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { EditProfileRequest } from "@api/user/[uid]/edit-profile/route.schema";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { useRouter } from "next/navigation";
+
+import { editProfile } from "@api/user/[uid]/edit-profile/route.client";
+
+interface editProfileParams {
+  uid: string;
+  firstName: string;
+  lastName: string;
+  pronouns: string;
+  email: string;
+}
+
+const EditProfileForm = (params: editProfileParams) => {
+  const [edited, setEdited] = React.useState(false);
+  type ValidationSchema = z.infer<typeof EditProfileRequest>;
```

Comment:
For readibility, I would move type declaration above `EditProfileForm` component.